portfolio/src/main/webapp/script.js

```diff
@@ -11,34 +11,45 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+$(document).ready(function () {
+  $("a").click(function () {
+    animateScrollTop($(this).attr("href").substr(1));
+  });
 
-async function getFormMessage() {
-  const response = await fetch('/data');
-  const jsonComments = await response.text();
-  const comments = JSON.parse(jsonComments);
-  for (const key in comments) {
-    for (const index in comments[key]) {
-      const node = document.createElement('p');
-      const textnode = document.createTextNode(comments[key][index].comment);
-      node.appendChild(textnode);
-      document.getElementById('comments-container').appendChild(node);
-    }
-  }
-}
+  $(".contact-form").submit(function (event) {
+    event.preventDefault();
+    const name = $(".form-name").val();
+    const comment = $(".form-comment").val();
+    postComment(comment, name);
+  });
+});
 
 const animateScrollTop = (targetClass) => {
-  $('html').animate({
-    scrollTop: $('.' + targetClass).offset().top
-  }, 2000);
+  $("html").animate(
+    {
+      scrollTop: $("." + targetClass).offset().top,
+    },
+    2000
+  );
 };
 
-$(document).ready(function() {
-  $('.contact-form').submit(function(event) {
-    event.preventDefault();
-    getFormMessage();
+async function postComment(comment, name) {
+  const data = { name: name, comment: comment };
+  const response = await fetch("/data", {
+    method: "POST",
+    body: JSON.stringify(data),
   });
+  const jsonComments = await response.text();
+  const comments = JSON.parse(jsonComments);
+  const commentNode = createNode("p", comments["comment"]);
+  document.getElementById("comments-container").appendChild(commentNode);
```

Comment:
alternatively you could just take advantage of jquery and do 
```
$('#comments-container').append(`<p>${comments.comment}</p>`)
```
and delete the `createNode` function
but it works so up to you

Reply:
In Servlet5 branch I have changed it to that 😁